src/decode.h

```diff
@@ -238,34 +238,56 @@ typedef uint16_t Port;
 #define PKT_IS_TOSERVER(p)  (((p)->flowflags & FLOW_PKT_TOSERVER))
 #define PKT_IS_TOCLIENT(p)  (((p)->flowflags & FLOW_PKT_TOCLIENT))
 
+typedef enum PacketAlertFlags {
+    /** flag to indicate the rule action (drop/pass) needs to be
+     * applied to the flow */
+    PACKET_ALERT_FLAG_APPLY_ACTION_TO_FLOW = BIT_U8(0),
+
+    /** alert was generated based on state */
+    PACKET_ALERT_FLAG_STATE_MATCH = BIT_U8(1),
+
+    /** alert was generated based on stream */
+    PACKET_ALERT_FLAG_STREAM_MATCH = BIT_U8(2),
+
+    /** alert is in a tx, tx_id set */
+    PACKET_ALERT_FLAG_TX = BIT_U8(3),
+
+    /** action was changed by rate_filter */
+    PACKET_ALERT_FLAG_RATE_FILTER_MODIFIED = BIT_U8(4),
+
+    /** alert is in a frame, frame_id set */
+    PACKET_ALERT_FLAG_FRAME = BIT_U8(5),
+
+    /** alert in a tx was forced */
+    PACKET_ALERT_FLAG_TX_GUESSED = BIT_U8(6),
+
+    /** accept should be applied to packet */
+    PACKET_ALERT_FLAG_APPLY_ACTION_TO_PACKET = BIT_U8(7),
+
+    /** user private flag 0 */
+    PACKET_ALERT_FLAG_USER0 = BIT_U8(12),
+
+    /** user private flag 1 */
+    PACKET_ALERT_FLAG_USER1 = BIT_U8(13),
+
+    /** user private flag 2 */
+    PACKET_ALERT_FLAG_USER2 = BIT_U8(14),
+
+    /** user private flag 3 */
+    PACKET_ALERT_FLAG_USER3 = BIT_U8(15),
+} PacketAlertFlags;
+
 /* structure to store the sids/gids/etc the detection engine
  * found in this packet */
 typedef struct PacketAlert_ {
     SigIntId num; /* Internal num, used for sorting */
     uint8_t action; /* Internal num, used for thresholding */
-    uint8_t flags;
+    uint16_t flags;
```

Comment:
There was a hole of 3 bytes here according to pahole, now there is a 1 byte hole.